libsolidity/codegen/ContractCompiler.cpp

```diff
@@ -183,10 +185,26 @@ size_t ContractCompiler::packIntoContractCreator(ContractDefinition const& _cont
 	m_context << deployRoutine;
 
 	solAssert(m_context.runtimeSub() != size_t(-1), "Runtime sub not registered");
+
+	ContractType contractType(_contract);
+	auto const& immutables = contractType.immutableVariables();
+	// Push all immutable values on the stack.
```

Comment:
I now changed this to use the stack as temporary storage for the immutable values.
This way I can avoid having to copy the code to a location after the reserved memory for the immutables.
So:
- after the constructor the immutables are in reserved memory
- then I read all of them onto the stack, so that I can discard that memory
- then I codecopy the code to memory location zero as before (I have the immutables on stack, so I can overwrite them in memory).
- then I assign the immutable values I had pushed onto the stack

This makes things much easier, since during ``assemble()`` of ``AssignImmutable`` I can assume that the code lives at memory offset zero instead of some non-constant higher memory location.

But it limits the number of stack slots all immutables combined can occupy to 1022 (otherwise I run out of stack here). But I'd say that's a sane limitation, what do you think?

Reply:
Yes, sounds good!